Commands

ConradSkelly opened the pull request:
## Please select the preveiw tab, then select the type of pull request you are making

- [Robot](?expand=1&template=robot_pull_request_template.md)
- [General](?expand=1&template=general_pull_request_template.md)
- [Draft](?expand=1&template=draft_template.md)

ConradSkelly committed:
started making the score commands and the intake command

ConradSkelly committed:
made the score l2 l3 commands and made the intake and reverse intake commands

ConradSkelly committed:
spotless applied

ConradSkelly committed:
fixed a error

eskiles commented on src/main/java/frc/robot/subsystems/Elevator.java:
```diff
@@ -145,6 +145,11 @@ public void periodic() {
     secondaryCurrent = Current.ofBaseUnits(secondaryMotor.getOutputCurrent(), Amps);
   }
 
+  public boolean getPosition(Distance desieredPostion) {
```
A more common naming convention for this type of function is to use "in range". The function compares a measured position and a target position to see if they are within a set tolerance. E.g. "positionIsInRange"

ConradSkelly committed:
tested code

JakeStew81 commented on src/main/java/frc/robot/commands/IntakeCommand.java:
```diff
@@ -0,0 +1,59 @@
+import frc.robot.subsystems.Intake;
+import frc.robot.subsystems.IntakePivot;
+
+public class IntakeCommand {
```
IntakeCommand*s*

JakeStew81 commented on src/main/java/frc/robot/commands/IntakeCommand.java:
```diff
@@ -0,0 +1,59 @@
+    return Commands.sequence(
+        elevator.setPositionCommand(ElevatorConstants.INTAKE_SETPOINT),
+        Commands.waitUntil(() -> elevator.getPosition(ElevatorConstants.INTAKE_SETPOINT)),
+        Commands.race(
```
This is going to end instantly, since a race ends when anything in it ends, and three of those end instantly. You want a normal parallel, which ends when all of them end

JakeStew81 commented on src/main/java/frc/robot/commands/IntakeCommand.java:
```diff
@@ -0,0 +1,59 @@
+            intake.setVeclocityCommand(IntakeConstants.INTAKE_VELOCITY),
+            intakePivot.moveIntakePivotCommand(IntakePivotConstants.OUT_POSITION),
+            indexer.setVelocityCommand(IndexerConstants.INDEX_VELOCITY),
+            endEffector.intakeCommand()));
```
Only the end effector stops itself

JakeStew81 commented on src/main/java/frc/robot/commands/ScoreCommand.java:
```diff
@@ -0,0 +1,30 @@
+import frc.robot.subsystems.Elevator;
+import frc.robot.subsystems.EndEffector;
+
+public class ScoreCommand {
```
ScoreCommand*s*

JakeStew81 commented on src/main/java/frc/robot/RobotContainer.java:
```diff
@@ -59,68 +58,35 @@ public RobotContainer() {
         swerve.driveFieldRelativeCommand(
             xboxController::getLeftY, xboxController::getLeftX, xboxController::getRightX));
+
+    elevator.setDefaultCommand(elevator.setPositionCommand(ElevatorConstants.INTAKE_SETPOINT));
```
When you call the set position command elsewhere, it ends almost instantly. A default command runs when no other commands are running on that subsystem...

JakeStew81 commented on src/main/java/frc/robot/subsystems/IntakePivot.java:
```diff
@@ -106,6 +106,15 @@ public void periodic() {
   }
 
+  public boolean atPosition() {
+    return (position.gte(IntakePivotConstants.OUT_POSITION.minus(Angle.ofBaseUnits(0.025, Radians)))
```
The 0.025 tolerance should be a constant

JakeStew81 commented on src/main/java/frc/robot/subsystems/Elevator.java:
```diff
@@ -145,6 +145,11 @@ public void periodic() {
   }
 
+  public boolean getPosition(Distance desieredPostion) {
+    return (position.gte(desieredPostion.minus(Meters.of(.025)))
```
The 0.025 tolerance should be a constant

JakeStew81 commented on src/main/java/frc/robot/commands/ScoreCommand.java:
```diff
@@ -0,0 +1,30 @@
+
+public class ScoreCommand {
+
+  public static Command levelThreeScore(Elevator elevator, EndEffector endEffector) {
```
Nitpick, but you don't really need the "Score" part, since that is implied by the fact that it is in score commands

ConradSkelly committed:
fixed issues

ConradSkelly committed:
spotless happend

ConradSkelly committed:
changed getPostion to PostionIsInRange

ConradSkelly committed:
compleated merge with dev and resolved conflicts

JakeStew81 commented on src/main/java/frc/robot/RobotContainer.java:
```diff
@@ -59,62 +52,26 @@ public RobotContainer() {
-    // Trigger spinIndexer = new Trigger(xboxController.a());
-    // spinIndexer.onTrue(indexer.indexerCommand(IndexerConstants.INDEX_VELOCITY));
-    // spinIndexer.onFalse(indexer.indexerCommand(0.0));
+    Trigger zero = xboxController.b();
```
The drivers did not ask for a zero gyro button

ConradSkelly committed:
changed nameing convention

ConradSkelly committed:
made a required change to merge

ConradSkelly committed:
Update IntakeCommands.java

Update IntakeCommands.java to repersetnt the accutate speeling of reverse instead of reves

JakeStew81 reviewed (approved):
Good work!

ConradSkelly committed:
spotless applied
